src/Mailer/Mailer.php

```diff
@@ -0,0 +1,222 @@
+    }
+
+    /**
+     * Returns the mailer's name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return str_replace('Mailer', '', join('', array_slice(explode('\\', get_class($this)), -1)));
+    }
+
+    /**
+     * Sets layout to use. Defaults to configured layout template if a custom layout
+     * could not be found.
+     *
+     * @param string $layout Name of the layout to use.
+     * @return $this object.
+     */
+    public function layout($layout)
+    {
+        $this->layout = $layout;
+        return $this;
+    }
+
+    /**
+     * Sets headers.
+     *
+     * @param array $headers Headers to set.
+     * @return $this object.
+     */
+    public function setHeaders(array $headers)
+    {
+        $this->_email->setHeaders($headers);
+        return $this;
+    }
+
+    /**
+     * Adds headers.
+     *
+     * @param array $headers Headers to set.
+     * @return $this object.
+     */
+    public function addHeaders(array $headers)
+    {
+        $this->_email->addHeaders($headers);
+        return $this;
+    }
+
+    /**
+     * Sets attachments.
+     *
+     * @param string|array $attachments String with the filename or array with filenames
+     * @return $this object.
+     * @throws \InvalidArgumentException
+     */
+    public function attachments($attachments)
+    {
+        $this->_email->attachments($attachments);
+        return $this;
+    }
+
+    /**
+     * Sets email view vars.
+     *
+     * @param string|array $key Variable name or hash of view variables.
+     * @param mixed $value View variable value.
+     * @return $this object.
+     */
+    public function set($key, $value = null)
+    {
+        $this->_email->viewVars(is_string($key) ? [$key => $value] : $key);
+        return $this;
+    }
+
+    /**
+     * Sends email.
+     *
+     * @param string $action The name of the mailer action to trigger.
+     * @param array $args Arguments to pass to the triggered mailer action.
+     * @param array $headers Headers to set.
+     * @return array
+     * @throws \Cake\Mailer\Exception\MissingActionException
+     * @throws \BadMethodCallException
+     */
+    public function send($action, $args = [], $headers = [])
+    {
+        if (!method_exists($this, $action)) {
+            throw new MissingActionException([
+                'mailer' => $this->getName() . 'Mailer',
+                'action' => $action,
+            ]);
+        }
+
+        $this->setHeaders($headers);
+
+        call_user_func_array([$this, $action], $args);
+
+        $result = $this->_email
+            ->profile((array)$this)
+            ->send();
+
+        $this->reset();
+        return $result;
+    }
+
+    /**
+     * Resets email instance to original config.
+     *
+     * @return $this object.
+     */
+    public function reset()
+    {
+        $this->_email->reset();
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
```

Comment:
I don't think apigen is going to be able to inherit docblocks for php's api docs :)